Adobe flavor of react-native-video (read-only)

bparadie opened the pull request:
I don't intend to ever merge this branch. This PR is just to document the delta between `brentvatne/react-native-video` and `bparadie/react-native-video#adobe`.
#### Delta between brentvatne/master and bparadie/adobe

That delta is getting bigger:
- removed redundant user data files
- `node_module/.../React.xcodeproj` now points to local `RCTVideo.m` etc.
- added support for `controls`: true => `AVPlayerViewController`, false => `AVPlayerLayer`
- renamed `seek` to `currentTime` (see https://github.com/brentvatne/react-native-video/issues/46)
- fixed `VideoPlayer.xcodeproj`: switched to `RCTWebSocket`
- replaced `Start/StopProgressTimer` hydraulics w/t `addPeriodicTimeObserverForInterval`
- set VideoPlayer's `getInitialState` for paused to `true`
#### HTML5 video element attributes

See https://www.w3.org/wiki/HTML/Elements/video

| Property | Status |
| --- | --- |
| `autoplay` | not implemented yet |
| `preload` | might not make sense |
| `controls` | implemented in this branch! |
| `loop` | `repeat` needs to be renamed to `loop` |
| `poster` | not implemented yet |
| `height` | not sure: I think w/h are set via styles |
| `width` | not sure: I think w/h are set via styles |
| `muted` | implemented! |
| `mediagroup` | probably doesn't make sense |
| `src` | implemented! |
| `volume` | implemented! |
| `currentTime` | implemented and renamed from `seek` |

PublicParadise committed:
Using AVPlayerViewController

PublicParadise committed:
Removing questionable peer dependency requirement

PublicParadise committed:
Forgot to include AVKit.h

PublicParadise committed:
Exporting createPlayerViewController

bparadie commented on RCTVideo.xcodeproj/project.pbxproj:
```diff
@@ -222,8 +222,8 @@
-					"$(SRCROOT)/../../React/**",
-					"$(SRCROOT)/../react-native/React/**",
+					"$(SRCROOT)/../../React",
+					"$(SRCROOT)/../react-native/React",
```
See http://stackoverflow.com/questions/3490629/error-argument-list-too-longrecursive-header-expansion-failed-at-application
It's still not pretty: I have to locally install `npm react-native` while leaving `package.json` unmodified.

PublicParadise committed:
Fix for Argument list too long

PublicParadise committed:
Adding support for AVPlayer onProgress

PublicParadise committed:
Merging master

PublicParadise committed:
Cleaning up formatting deltas

PublicParadise committed:
Fixing VideoPlayer.xcodeproj: switched to RCTWebSocket

PublicParadise committed:
Let's delete Brent's user data

bparadie commented on Examples/VideoPlayer/.gitignore:
```diff
@@ -1 +1,3 @@
 node_modules/**/*
+VideoPlayer.xcodeproj/project.xcworkspace/**/*
+VideoPlayer.xcodeproj/xcuserdata/**/*
```
Those should not be checked in.

bparadie commented on Examples/VideoPlayer/node_modules/react-native-video/RCTVideo.xcodeproj/xcuserdata/brentvatne.xcuserdatad/xcschemes/RCTVideo.xcscheme:
```diff
@@ -1,77 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
This file got probably checked in by accident.

bparadie commented on Examples/VideoPlayer/node_modules/react-native-video/RCTVideo.xcodeproj/xcuserdata/brentvatne.xcuserdatad/xcschemes/xcschememanagement.plist:
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
This file got probably checked in by accident.

PublicParadise committed:
Letting node_modules/.../RCTVideo.xcodeproj point to the current sources in this branch

bparadie commented on Examples/VideoPlayer/node_modules/react-native-video/RCTVideo.xcodeproj/project.pbxproj:
```diff
@@ -25,10 +25,10 @@
+		BBD49E391AC8DEF000610F8E /* RCTVideo.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = RCTVideo.h; path = ../../../../RCTVideo.h; sourceTree = "<group>"; };
+		BBD49E3A1AC8DEF000610F8E /* RCTVideo.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; name = RCTVideo.m; path = ../../../../RCTVideo.m; sourceTree = "<group>"; };
+		BBD49E3B1AC8DEF000610F8E /* RCTVideoManager.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = RCTVideoManager.h; path = ../../../../RCTVideoManager.h; sourceTree = "<group>"; };
+		BBD49E3C1AC8DEF000610F8E /* RCTVideoManager.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; name = RCTVideoManager.m; path = ../../../../RCTVideoManager.m; sourceTree = "<group>"; };
```
The only reason why I could see @brentvatne check in a `RCTVideo.xcodeproj` under `node_modules` is for letting the source files point to the local implementation and not the one we pull from `npm`. But to my surprise I found the source file paths pointing to the pulled source. I'm gonna go with my instinct here and let those point to the local sources...

bparadie commented on RCTVideo.m:
```diff
@@ -18,8 +17,8 @@ @implementation RCTVideo
   AVPlayerItem *_playerItem;
+  AVPlayerViewController *_playerLayer;
   BOOL _playerItemObserverSet;
-  AVPlayerLayer *_playerLayer;
```
I am planning on cleaning this up as follows: 
- I'll introduce a new property called `controls` that mirrors the `video` tag's property (http://www.w3schools.com/tags/att_video_controls.asp)
- If `controls` is set I'll use `AVPlayerViewController` otherwise I'll use `AVPlayerLayer`
  That would make the most sense to me.

bparadie commented on RCTVideo.m:
```diff
@@ -238,12 +294,21 @@ - (void)setPaused:(BOOL)paused
 }
 
+
 - (void)setSeek:(float)seekTime
```
Shouldn't `seek` be renamed to `currentTime`, see http://www.w3schools.com/tags/av_prop_currenttime.asp ?

PublicParadise committed:
Bringing the code closer back to the original

PublicParadise committed:
Adding support for 'controls'

PublicParadise committed:
Fixes layoutSubviews

PublicParadise committed:
Making setControls more robust

PublicParadise committed:
Consolidating setSrc w/t setControls

PublicParadise committed:
Replacing Start/StopProgressTimer hydraulics w/t addPeriodicTimeObserverForInterval

bparadie commented on RCTVideo.m:
```diff
@@ -126,16 +155,26 @@ - (void)setSrc:(NSDictionary *)source
+                     ^(CMTime time)
+                     {
+                         [weakSelf sendProgressUpdate];
+                     }];
```
The `startProgressTimer` / `stopProgressTimer` hydraulics seemed unnecessary. I find `addPeriodicTimeObserverForInterval` so much easier to use and clear.

bparadie commented on RCTVideo.m:
```diff
@@ -188,13 +227,12 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
       [_eventDispatcher sendInputEventWithName:RNVideoEventLoadingError body:@{
         @"error": @{
-          @"code": [NSNumber numberWithInt:_playerItem.error.code],
+          @"code": [NSNumber numberWithInteger:_playerItem.error.code],
```
That was triggering a warning.

PublicParadise committed:
Fixing seek, replacing with currentTime

PublicParadise committed:
VideoPlayer's getInitialState for paused is of course 'true', because autoPlay is 'false'

PublicParadise committed:
Fixing validAttributes

PublicParadise committed:
Switching to PlayerViewController in insertReactSubview if necessary

brentvatne commented:
I'll keep an eye on this, looks neat :smile:

bparadie commented:
@brentvatne Oh, thanks for watching! I think, `addPeriodicTimeObserverForInterval` would be a great improvement to `react-native-video`.

Commit:
Bump version for peerDependencies, supporting upcoming 0.5.0

brentvatne commented:
@bparadie - could you submit a PR for that by any chance? Have a lot on my plate right now and that would help :smile:

brentvatne committed:
Remove annoying peerDependencies for now

isair committed:
Add playableDuration property to onProgress event

isair committed:
Improve calculatePlayableDuration documentation

Commit:
support for absolute path and file:// paths

brentvatne committed:
Merge pull request #52 from isair/feature/playable-duration

Add playableDuration property to onProgress event

brentvatne committed:
Merge pull request #54 from machard/feature/local-files

support for absolute path and file:// paths

markburns committed:
Fix setting of resizeMode and repeat when changing source

Commit:
Fix whitespace

Commit:
Bump version

isair committed:
Fix 'Invalid number value (NaN) in JSON write' crash

brentvatne committed:
Merge pull request #56 from isair/fix/playable-duration-edge-cases

Fix 'Invalid number value (NaN) in JSON write' crash

brentvatne committed:
Bump version

isair committed:
Fix attachListeners method's threading issue & other notification handling issues

markburns committed:
Fix inline image in README

brentvatne committed:
Merge pull request #63 from markburns/patch-2

Fix inline image in README

brentvatne committed:
Various fixes
- Call super in removeFromSuperview
- setRate has to come at the end of modifiers
- Switch to requireNativeComponent

NishanthShankar committed:
Changes to README

brentvatne committed:
Update for npm

brentvatne committed:
Update package.json

brentvatne committed:
0.4.7

brentvatne committed:
Remove RCTWebSocketDebugger from Example project

brentvatne committed:
Pause video when the app is backgrounded

brentvatne committed:
Merge pull request #64 from NishanthShankar/master

Changes to README

Commit:
Remove Notification Center observers on dealloc

brentvatne committed:
Merge pull request #65 from jhabdas/master

Remove Notification Center observers on dealloc

brentvatne committed:
Bump version

PublicParadise committed:
Play/Pause events

PublicParadise committed:
Merge branch 'adobe' of https://github.com/bparadie/react-native-video into adobe

PublicParadise committed:
Hand merged 0.4.9

PublicParadise committed:
Merging 0.4.9

PublicParadise committed:
Excluding VideoPlayer.xcodeproj xcuserdata, project.xcworkspace

PublicParadise committed:
Merged master
